src/main/java/uk/gov/laa/ccms/caab/mapper/ClientDetailMapper.java

```diff
@@ -0,0 +1,168 @@
+package uk.gov.laa.ccms.caab.mapper;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+import uk.gov.laa.ccms.caab.bean.ClientDetails;
+import uk.gov.laa.ccms.soa.gateway.model.ClientDetail;
+
+/**
+ * Maps between Client details form bean and the soa-api client detail. Requires the
+ * {@link uk.gov.laa.ccms.caab.mapper.IgnoreUnmappedMapperConfig}.
+ */
+@Mapper(componentModel = "spring", config = IgnoreUnmappedMapperConfig.class)
+public interface ClientDetailMapper {
+
+  @Mapping(target = "details.name.title",
```

Comment:
This is one hell of a mapping method. Could it be broken down a bit?
Eg. a method to convert from ClientDetails to NameDetail.
And another to handle ClientDetails to ClientPersonalDetail.

Then reference them all in this method if needed.

Reply:
Will break it down

Reply:
amended